### PR TITLE
VMO-4347 Fix for expression editor bug

### PR DESCRIPTION
**Pull request by poojahpatil90**
Note to reviewers: I have removed some legacy code in `resource-variant-text-editor` component. My thinking here was that it used `floip-parser` which we wanted to get rid of and there was code to count characters, autogen etc which were not working. Let me know your thoughts if you feel those pieces need to be there as commented code.

### CODE REVIEW

File: src/components/interaction-designer/block/BlockToolbar.vue [hunk 1]
```diff
@@ -113,6 +113,10 @@ class BlockToolbar extends mixins(Lang) {
     this.$router.replace({
       name: routerName,
       params: {blockId: this.block.uuid},
+    }).catch((err) => {
+      if (err.name !== 'NavigationDuplicated') {
```
Comment (safydy-r) [inline src/components/interaction-designer/block/BlockToolbar.vue:117]: Thanks for fixing it

Comment (poojahpatil90) [reply src/components/interaction-designer/block/BlockToolbar.vue:117]: Your welcome 👍

File: src/components/interaction-designer/resource-editors/ResourceVariantTextEditor.vue [hunk 2]
```diff
@@ -1,300 +1,57 @@
...
+@Component({
+  components: {ExpressionInput},
+})
+class AdvancedExitEditor extends mixins(Lang) {
+  @Prop() readonly resourceId!: IResource['uuid']
+  @Prop({default: ''}) readonly label!: string
+  @Prop() readonly placeholder!: string
+  @Prop() readonly resourceVariant!: IResourceValue
+  @Prop() readonly mode!: string
+  @Prop({default: true}) readonly enableAutogenButton!: boolean
```
Comment (safydy-r) [inline src/components/interaction-designer/resource-editors/ResourceVariantTextEditor.vue:32]: I think this is unused now, we can remove it. Then update the component usage by removing lines like this https://github.com/FLOIP/flow-builder/blob/master/src/components/interaction-designer/resource-editors/ResourceEditor.vue#L38

File: src/components/interaction-designer/resource-editors/ResourceVariantTextEditor.vue [hunk 3]
```diff
@@ -1,300 +1,57 @@
...
+  @Prop({default: 2}) readonly rows!: number
+
+  get content(): string {
+    return this.resourceVariant.value
   }
 
-  .content-toolbar {
-    margin-top: 1px;
+  commitExpressionChange(value: string): void {
+    const {resourceId, mode} = this
```
Comment (safydy-r) [inline src/components/interaction-designer/resource-editors/ResourceVariantTextEditor.vue:40]: If we used destruction approach for resourceId, mode ... I'd suggest we put `resourceVariant` here as well. Then change `this.resourceVariant` with `resourceVariant` for all below codes

File: src/components/interaction-designer/resource-editors/ResourceVariantTextEditor.vue [hunk 4]
```diff
@@ -1,300 +1,57 @@
...
+import {IResource, IResourceValue} from '@floip/flow-runner'
+import {namespace} from 'vuex-class'
+import ExpressionInput from '@/components/common/ExpressionInput.vue'
+
+const flowVuexNamespace = namespace('flow')
+
+@Component({
+  components: {ExpressionInput},
+})
+class AdvancedExitEditor extends mixins(Lang) {
```
Comment (safydy-r) [inline src/components/interaction-designer/resource-editors/ResourceVariantTextEditor.vue:26]: Thanks @poojahpatil90 for refactoring this into TS version 💯

File: src/components/interaction-designer/resource-editors/ResourceVariantTextEditor.vue [hunk 5]
```diff
@@ -1,300 +1,57 @@
 <template>
   <div class="resource-variant-text-editor">
-    <div
```
Comment (safydy-r) [inline src/components/interaction-designer/resource-editors/ResourceVariantTextEditor.vue:3]: Thanks for thinking about this.
To be honest, I don't have a concern about removing this; but I'm afraid I don't have enough context about expected behaviour here. So I'd like to wait for @bzabos for confirmation.

File: src/components/interaction-designer/resource-editors/ResourceVariantTextEditor.vue [hunk 6]
```diff
@@ -1,300 +1,57 @@
...
-      </div>
-    </div>
-
-    <!--    <template v-if="!isEditable">-->
-    <!--      <p v-if="content">{{content}}</p>-->
-    <!--      <p>-->
-    <!--        <em class="text-muted">{{'flow-builder.no-sms-content-yet' | trans}}</em>-->
-    <!--      </p>-->
-    <!--    </template>-->
+    <expression-input
```
Comment (safydy-r) [inline src/components/interaction-designer/resource-editors/ResourceVariantTextEditor.vue:3]: btw @poojahpatil90 , do we have a future ticket to fix the autocompletion being far from the input ?
Screen Shot 2021-07-30 at 2 40 02 PM

**Review by safydy-r — COMMENTED** [review]
Thanks @poojahpatil90 , I'd like to leave the final decision for @bzabos because there is this part of code you have removed.

File: src/components/interaction-designer/resource-editors/ResourceVariantTextEditor.vue (hunk 2 above)
Comment (poojahpatil90) [reply src/components/interaction-designer/resource-editors/ResourceVariantTextEditor.vue:32]: Your right!  I removed this unused bit.

File: src/components/interaction-designer/resource-editors/ResourceVariantTextEditor.vue (hunk 3 above)
Comment (poojahpatil90) [reply src/components/interaction-designer/resource-editors/ResourceVariantTextEditor.vue:40]: Done!

File: src/components/interaction-designer/resource-editors/ResourceVariantTextEditor.vue (hunk 6 above)
Comment (poojahpatil90) [reply src/components/interaction-designer/resource-editors/ResourceVariantTextEditor.vue:3]: Yes, here is the ticket - https://viamoinc.atlassian.net/browse/VMO-4217
I updated the description.

Comment (safydy-r) [reply src/components/interaction-designer/resource-editors/ResourceVariantTextEditor.vue:3]: Nice, thanks

**Comment by safydy** [comment]
@bzabos , flagging you here again, for follow up. I made a review, overall I'm good, but there is a change in `src/components/interaction-designer/resource-editors/ResourceVariantTextEditor.vue` I'd like to have approval from you. thanks

**Review by bzabos — APPROVED** [review]
@poojahpatil90 Looks good! Thanks for doing cleanup as you go, this is really nice to see.

**Comment by bzabos** [comment]
@poojahpatil90 Can you touch base with @jontam to communicate that we no longer have the character count feature? This could be important from the perspective of "feature parity" with Tree Builder. Thanks!

**Pull request merged**